### ISSUE TITLE
UuidMatcher provides a Generator even when a value is specified

### ISSUE THREAD

**Issue by jon-ruckwood** [DiUS/pact-jvm#598]
When specifying an example UUID using the Groovy `uuid` matcher, e.g.:

```groovy
import au.com.duis.pact.consumer.groovy.Matchers

// omitted...

id Matchers.uuid('88aee6ad-bff6-4e11-ae5c-626f00ad13a0') 
```

I expect the same example value to be present in the generated Pact file and the json returned in the HTTP response.

However, whilst the example value is written to the Pact file, a different value is returned in the HTTP response.

I believe the problem is in [`UuidMatcher::getGenerator`][0]. I think the implementation should be:

```groovy
Generator getGenerator() {
    value == null ? new UuidGenerator() : null
}
```

I was planning on submitting a Pull Request, but I had trouble building the project.

[0]: https://github.com/DiUS/pact-jvm/blob/master/pact-jvm-consumer-groovy/src/main/groovy/au/com/dius/pact/consumer/groovy/UuidMatcher.groovy#L17-L19

**Comment by uglyog** [comment]
You are correct, that should be changed. I'll update it for you.

**Comment by uglyog** [comment]
Fix has been released with version 3.5.11
